src/lightning/pytorch/strategies/launchers/xla.py

```diff
@@ -104,6 +103,18 @@ def _wrapping_function(
         return_queue: SimpleQueue,
         global_states: Optional[_GlobalStateSnapshot] = None,
     ) -> None:
+        import torch_xla.core.xla_model as xm
+        from torch_xla.experimental import pjrt
+
+        if pjrt.using_pjrt() and len(xm.get_xla_supported_devices()) > 1:
```

Comment:
`trainer.num_devices` will use `strategy.parallel_devices`. If the user does `Trainer(devices=[0])` then it will return the correct number (1) vs `get_xla_supported_devices()` which would be 4

```suggestion
        if pjrt.using_pjrt() and trainer.num_devices > 1:
```

Reply:
So in the case of a distributed job with `XLAStrategy` using all the devices, what would the `trainer.num_devices` be?
On V2 and V3, `len(xm.get_xla_supported_devices())` would be 2 as they see 2 logical devices in the same process (used by 2 threads), and deepcopy is needed. On V4 it would be 1, so no need to deepcopy.

Reply:
Oh ignore my comment. I didn't notice that `get_xla_supported_devices` returns a single value when run in the spawned process